backend/rdp/pointer.c

```diff
@@ -0,0 +1,33 @@
+#include <stdlib.h>
+#include <wayland-server.h>
+#include <wlr/types/wlr_input_device.h>
+#include <wlr/types/wlr_pointer.h>
+#include <wlr/interfaces/wlr_pointer.h>
+#include <wlr/interfaces/wlr_input_device.h>
+#include <wlr/util/log.h>
+#include "backend/rdp.h"
+#include "util/signal.h"
+
+static struct wlr_input_device_impl input_device_impl = { 0 };
+
+struct wlr_rdp_input_device *wlr_rdp_pointer_create(
+		struct wlr_rdp_backend *backend) {
+	struct wlr_rdp_input_device *device =
+		calloc(1, sizeof(struct wlr_rdp_input_device));
+
+	int vendor = 0;
+	int product = 0;
+	const char *name = "rdp";
+	struct wlr_input_device *wlr_device = &device->wlr_input_device;
+	wlr_input_device_init(wlr_device, WLR_INPUT_DEVICE_POINTER,
+			&input_device_impl, name, vendor, product);
+
+	if (!(wlr_device->pointer = calloc(1, sizeof(struct wlr_pointer)))) {
+		wlr_log(WLR_ERROR, "Failed to allocate RDP pointer device");
+		return NULL;
+	}
+	wlr_pointer_init(wlr_device->pointer, NULL);
```

Comment:
Right before, you can map the pointer to the output. See e.g. how the Wayland backend does it.

```c
wlr_dev->output_name = strdup(output->wlr_output.name);
```